File: java/tp2/src/JogoAdivinhacao.java
import java.util.Scanner;
import java.util.Random;

public class JogoAdivinhacao {
    public static void main(String[] args) {

        Scanner scanner = new Scanner(System.in);
        Random random = new Random();

        // Gera um número aleatório entre 1 e 100
        int numeroSecreto = random.nextInt(100) + 1;

        int palpite;

        System.out.println("===== JOGO DE ADIVINHAÇÃO =====");
        System.out.println("Tente adivinhar o número entre 1 e 100!");

        // Laço de repetição
        do {
            System.out.print("Digite seu palpite: ");
            palpite = scanner.nextInt();

            if (palpite < numeroSecreto) {
                System.out.println("O número secreto é MAIOR. Tente novamente!");

            } else if (palpite > numeroSecreto) {
                System.out.println("O número secreto é MENOR. Tente novamente!");

            } else {
                System.out.println("\nParabéns! Você acertou!");
                System.out.println("O número secreto era: " + numeroSecreto);
            }

        } while (palpite != numeroSecreto);

        scanner.close();
    }
}

File: csharp/TP2-c#/src/JogoAdivinhacao.cs
using System;

class Program
{
    static void Main()
    {
        Random random = new Random();

        int numeroSecreto = random.Next(1, 101);
        int palpite = 0;
        int tentativas = 0;

        Console.WriteLine("===== JOGO DE ADIVINHAÇÃO =====");
        Console.WriteLine("Tente adivinhar um número entre 1 e 100!");

        while (palpite != numeroSecreto)
        {
            Console.Write("Digite seu palpite: ");
            palpite = int.Parse(Console.ReadLine());

            tentativas++;

            if (palpite < numeroSecreto)
            {
                Console.WriteLine("O número secreto é maior!");
            }
            else if (palpite > numeroSecreto)
            {
                Console.WriteLine("O número secreto é menor!");
            }
            else
            {
                Console.WriteLine();
                Console.WriteLine("Parabéns! Você acertou!");
                Console.WriteLine($"Número de tentativas: {tentativas}");
            }
        }
    }
}
